subprojects/core-api/src/main/java/org/gradle/api/Project.java

```diff
@@ -1583,6 +1584,16 @@ public interface Project extends Comparable<Project>, ExtensionAware, PluginAwar
     @Incubating
     DependencyFactory getDependencyFactory();
 
+    /**
+     * Provides access to methods to create various kinds of {@link org.gradle.api.artifacts.DependencyConstraint DependencyConstraint}
+     * instances.
+     *
+     * @return the dependency constraint factory. Never returns null.
+     * @since 8.7
+     */
+    @Incubating
+    DependencyConstraintFactory getDependencyConstraintFactory();
```

Comment:
Yeah I have the same question. Ditto for the above method. 
Why are these on `Project`? Can we just allow users to inject these when they want them?

Reply:
I removed it. I think we can remove the general dependency factory one as well, in a separate PR. Originally I think they were added for usage in mapping providers in a build script, but you can get one from `Dependencies` so that's good enough.